File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewobject/ShopMap.java
package com.dewonderstruck.apps.ashx0.viewobject;

import androidx.annotation.NonNull;
import androidx.room.Entity;

/**
 * Created by Vamsi Madduluri on 3/19/19.
 * Contact Email : vamsii.wrkhost@gmail.com
 */

@Entity(primaryKeys = "id")
public class ShopMap {

    @NonNull
    public final String id;

    public final String mapKey;

    public final String productId;

    public final int sorting;

    public final String addedDate;

    public ShopMap(@NonNull String id, String mapKey, String productId, int sorting, String addedDate) {
        this.id = id;
        this.mapKey = mapKey;
        this.productId = productId;
        this.sorting = sorting;
        this.addedDate = addedDate;
    }
}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewobject/ShopMap.kt
package com.dewonderstruck.apps.ashx0.viewobject

import androidx.room.Entity

/**
 * Created by Vamsi Madduluri on 3/19/19.
 * Contact Email : vamsii.wrkhost@gmail.com
 */
@Entity(primaryKeys = ["id"])
class ShopMap(val id: String, val mapKey: String, val productId: String, val sorting: Int, val addedDate: String)